astropy/table/tests/test_pprint.py

```diff
@@ -316,6 +316,15 @@ def test_pprint_all(self, table_type, capsys):
         assert len(out) == BIG_WIDE_ARR.shape[0] + 3
 
 
+class TestPprintColumn:
+    @pytest.mark.parametrize("scalar", [1, u.Quantity(0.6, "eV")])
+    @pytest.mark.parametrize("show_dtype", [True, False])
+    def test_pprint_scalar(self, scalar, show_dtype):
+        # see https://github.com/astropy/astropy/issues/12584
+        c = Column(scalar)
+        c.pprint(show_dtype=show_dtype)
```

Comment:
Note that I am deliberately not checking what's actually printed because I think it may be incorrect at the moment _for reasons that go beyond the scope of the PR_.
Specifically I'm looking at https://github.com/astropy/astropy/blob/7a7b6ba6a1fc65aeb4585819d3c4e50b97631dda/astropy/table/column.py#L1279-L282, which effectively forces scalar Columns to print like pure numerical scalars (leaving the unit out if `data` is a `Quantity` !). I'll open a separate issue for this.

Reply:
I think it would be valuable to see the scalar aspect of the printing, even if the unit is wrong.

Reply:
I'm with @nstarman here, we should test the actual output, especially now that I think we have a more permanent solution.

Reply:
If I understand correctly https://github.com/astropy/astropy/issues/15754 should be closed as "not a bug" and the behaviour, as of the current state of this branch, should be made the baseline of the test ?

Reply:
I think for `.pprint` , we can call the current behaviour fine. But #15754 is about `repr`, which I'm not sure about. Since it is not addressed here, I think we should just leave it open.

Reply:
Agreed on testing the output. In fact this whole issue has been focused on `pprint` but really the problem is in `pformat`. I would not worry about `show_dtype` here and just have the two cases of `1` and `1.0 eV`. (There is no need here to test formatting of the Quantity itself, so it helps in testing to pick values that have an exact floating point repr).

In a lot of tests I use a pattern like:
```
c = Column(scalar, name="a")
out = c.pformat()
exp = [' a ', '---', '  1']
assert out == exp
```
This way if there are diffs then running pytest with `-vv` will show what's going on. You can parametrize the `scalar` and `exp` values.

Reply:
done !